nekoyume/Assets/_Scripts/UI/Module/Button/ConditionalButton.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.ComponentModel;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Nekoyume.UI.Module
+{
+    using Nekoyume.L10n;
+    using UniRx;
+
+    public class ConditionalButton : MonoBehaviour
+    {
+        [SerializeField]
+        private GameObject normalObject = null;
+
+        [SerializeField]
+        private GameObject conditionalObject = null;
+
+        [SerializeField]
+        private GameObject disabledObject = null;
+
+        [SerializeField]
+        private string conditionInfoKey = null;
+
+        public System.Action OnClick { protected get; set; }
+
+        private Button _activatedButton = null;
+        private Func<bool> _conditionFunc = null;
+        private IDisposable _onClickDisposable = null;
+        private bool _interactable = true;
+
+        public bool Interactable
+        {
+            get => _interactable;
+            set
+            {
+                _interactable = value;
+                UpdateObjects();
+            }
+        }
+
+        private void OnEnable()
+        {
+            UpdateObjects();
+        }
+
+        private void OnDisable()
+        {
+            _onClickDisposable?.Dispose();
+        }
+
+        private void OnDestroy()
+        {
+            _onClickDisposable?.Dispose();
+        }
```

Comment:
It's enough call once in `OnDisable()` method.

```suggestion
        private void OnDisable()
        {
            _onClickDisposable?.Dispose();
        }
```